src/mscorlib/src/System/String.cs

```diff
@@ -666,6 +666,37 @@ private unsafe String CtorCharPtrStartLength(char* ptr, int startIndex, int leng
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         public extern String(char c, int count);
 
+        private unsafe string CtorCharRefLength(ref char value, int length)
+        {
+            Debug.Assert(length >= 0);
+
+            if (length == 0)
+            {
+                return Empty;
+            }
+
+            try
+            {
+                string result = FastAllocateString(length);
+                fixed (char* dest = &result._firstChar, src = &value)
+                {
+                    wstrcpy(dest, src, length);
+                }
+                return result;
+            }
+            catch (NullReferenceException)
+            {
+                throw new ArgumentOutOfRangeException(nameof(value), SR.ArgumentOutOfRange_PartialWCHAR);
+            }
+        }
+
+        public String(ReadOnlySpan<char> value) :
```

Comment:
I do not think that this will work. All string constructors are special - they have to be intrinsics. The implementation worker method has to match the public signature.